app/src/main/java/com/realityexpander/tasky/presentation/login_screen/LoginScreen.kt

```diff
@@ -1,22 +1,174 @@
+            isError = loginState.isInvalidPassword,
+            onPasswordChange = {
+                scope.launch {
+                    viewModel.onEvent(LoginEvent.UpdatePassword(it))
+                }
+            },
+            isPasswordVisible = loginState.isPasswordVisible,
+            clickTogglePasswordVisibility = {
+                viewModel.sendEvent(
+                    LoginEvent.TogglePasswordVisibility(loginState.isPasswordVisible)
+                )
+            },
+            imeAction = ImeAction.Done,
+            doneAction = {
+                performLogin()
+            }
+        )
+        if(loginState.isInvalidPassword) {
+            Text(text = "Invalid password", color = Color.Red)
+            Spacer(modifier = Modifier.height(8.dp))
+        }
+
+        // LOGIN BUTTON
+        Spacer(modifier = Modifier.height(16.dp))
+        Button(onClick = {
+                performLogin()
+            },
+            enabled = !loginState.isLoading,
+            modifier = Modifier
+                .align(alignment = Alignment.End)
+        ) {
+            Text(text = "Login")
+            if(loginState.isLoading) {
+                CircularProgressIndicator(
+                    modifier = Modifier
+                        .padding(start = 4.dp)
+                        .size(16.dp)
+                        .align(alignment = CenterVertically)
+                )
+            }
+        }
+        Spacer(modifier = Modifier.height(32.dp))
+
+        // REGISTER BUTTON
+        Text(text = "Not a member? Sign up",
+            color = Color.Cyan,
+            modifier = Modifier
+                .align(alignment = Alignment.CenterHorizontally)
+                .clickable(onClick = {
+                    navigator.clearBackStack("LoginScreenDestination")
+                    navigator.navigate(
+                        RegisterScreenDestination(
+                            email = loginState.email,
+                            password = loginState.password,
+                            confirmPassword = confirmPassword
+                        )
+                    )
+                })
+        )
+        Spacer(modifier = Modifier.height(16.dp))
+
+//        Text(text = "Forgot password?",
+//            color = Color.Cyan,
+//            modifier = Modifier
+//                .align(alignment = Alignment.CenterHorizontally)
+//        )
+//        Spacer(modifier = Modifier.height(16.dp))
+
+        // STATUS //////////////////////////////////////////
+
+        if(loginState.isError) {
+            Text(
+                text = "Error: ${loginState.errorMessage}",
+                color = Color.Red,
+            )
+            Spacer(modifier = Modifier.height(8.dp))
+        }
+        if(loginState.isLoggedIn) {
+            Text(text = "Logged in!")
+            Spacer(modifier = Modifier.height(8.dp))
+        }
+        if(loginState.statusMessage.isNotEmpty()) {
```

Comment:
I personally prefer `null` values if something is not existent (that's exactly what nulls are for). Having a non-null string that is empty doesn't make it fully clear if there is a status message or not.

The only scenario where I use empty strings as default values would be text field values, since in that case it does make sense to me to have an empty string.

Reply:
OK this makes sense. I will convert these "no message" strings to nullables, and check for `null` instead of `isNotEmpty` 

😀